doc/storagemond/storagemond-hld.md

```diff
@@ -0,0 +1,452 @@
+fs_io_writes            = STRING                    ; Describes the total number of filesystem writes completed successfully                 (Dynamic)
+current_fs_io_reads     = STRING                    ; Describes the latest filesystem reads completed successfully                           (Dynamic)
+current_fs_io_writes    = STRING                    ; Describes the latest filesystem writes completed successfully                          (Dynamic)
+fs_io_writes            = STRING                    ; Describes the total number of filesystem writes completed successfully                 (Dynamic)
+disk_io_reads           = STRING                    ; Describes the total number of reads completed successfully from the SSD (Bytes)        (Dynamic)
+disk_io_writes          = STRING                    ; Describes the total number of writes completed on the SSD (Bytes)                      (Dynamic)
+reserved_blocks         = STRING                    ; Describes the reserved blocks count of the SSD                                         (Dynamic)
+firmware                = STRING                    ; Describes the Firmware version of the SSD                                              (Dynamic)
+health                  = STRING                    ; Describes the overall health of the SSD as a % value based on several SMART attrs      (Dynamic)
+```
+
+NOTE: disk_io_reads and disk_io_writes return total LBAs read/written. 'LBA' stands for Logical Block Address. 
+To get the raw value in bytes, we multiply thr num. LBAs by the disk's logical block address size (typically 512 bytes).<br>
+
+Example: For an SSD with name 'sda', the STATE_DB entry would be:
+
+```
+root@sonic:~# docker exec -it database bash
+root@sonic:/# redis-cli -n 6
+127.0.0.1:6379[6]> keys STORAGE*
+1) "STORAGE_INFO|mmcblk0"
+2) "STORAGE_INFO|sda"
+127.0.0.1:6379[6]>
+127.0.0.1:6379[6]> hgetall STORAGE_INFO|sda
+ 1) "device_model"
+ 2) "SATA SSD"
+ 3) "serial"
+ 4) "SPG2043056Z"
+ 5) "firmware"
+ 6) "FW1241"
+ 7) "health"
+ 8) "N/A"
+ 9) "temperature"
+10) "30"
+11) "fs_io_reads"
+12) "28753"
+13) "fs_io_writes"
+14) "92603"
+15) "disk_io_reads"
+16) "15388141951"
+17) "disk_io_writes"
+18) "46070618960"
+19) "reserved_blocks"
+20) "32"
+
+```
+
+### **3.2 FS Stats Sync `STATE_DB` Schema**
+```
+; Defines information for FS Stats synchronization
+
+key                     = STORAGE_INFO|FSSTATS_SYNC     ; This key is for information pertaining to synchronization of FSIO Reads/Writes
+
+; field                 = value
+
+successful_sync_time    = STRING                        ; The latest successful sync time of FSIO reads and writes to file in '+%Y-%m-%d::%H:%M:%S' format
+
+```
+
+### **3.3 ConfigDB Schema**
+```
+; Defines information for stormon config
+
+key                        = STORAGEMOND_CONFIG   ; This key is for information about a stormon daemon configuration
```

Comment:
Can you please add yang schema to the HLD? If we are modifying config_db its mandatory to add a section for yang

Reply:
Done in latest commit.